### PR TITLE
Add support for lists in the openapi specification

### PR DESCRIPTION
**Pull request by AWinterman**
The pets example has lists in it, which could not render into openapi due to lack of support for slice types.

### CODE REVIEW

**Comment by masseelch** [comment]
Hey @AWinterman, great catch.

**Comment by AWinterman** [comment]
@masseelch done!

File: spec/spec.go [hunk 1]
```diff
@@ -98,8 +98,9 @@ type (
 		Example  interface{} `json:"example,omitempty"`
 	}
 	Type struct {
-		Type   string `json:"type"`
-		Format string `json:"format,omitempty"`
+		Type   string      `json:"type,omitempty"`
+		Format string      `json:"format,omitempty"`
+		Items  interface{} `json:"items,omitempty"` // of type `Type`, but recursive types are not valid.
```
Comment (masseelch) [inline spec/spec.go:103]:
```suggestion
		Items  *Type `json:"items,omitempty"`
```

**Review by masseelch — APPROVED** [review]
Thank you very much @AWinterman

File: spec/spec.go (hunk 1 above)
Comment (masseelch) [reply spec/spec.go:103]: If you make the it a pointer recursion is allowed ;-)

**Pull request merged**